lib/ContainerManager.js

```diff
@@ -146,6 +151,32 @@ Server.prototype.configure = function(config, done) {
   done();
 };
 
+Server.prototype.jsonWebTokenParser = function(publicKey) {
+  var handleError = function(res) {
+    res.status(401);
+    res.json({error: 'Invalid JWT provided'});
+  };
+  return function(req, res, next) {
+    req.log.info(req.headers);
+    if (req.headers && req.headers.authorization) {
+      var parts = req.headers.authorization.split(' ');
+      if (parts.length === 2 && parts[0] === 'Bearer') {
+        var token = parts[1];
+      }
+      else {
+        handleError(res);
+      }
+    }
+    jwt.verify(token, publicKey, {algorithms: ['RS256']}, function(error, data) {
```

Comment:
We *could*, I assume make that easily configurable. But of the signing algorithms supported by this library, only this one uses Public/Private keys. So I guess the question is "should we support the use of arbitrary key strings." I'd argue that internally, we should not support that. Externally, since this is an open source project, we may want to support arbitrary keys but I'd suggest that if a user wants to use JWT to tighten things up, they probably wouldn't find this to be a burden.